pkg/config/kubeClient.go

```diff
@@ -54,17 +54,22 @@ func getKubeConfig() *rest.Config {
 
 // Get kubernetes client for discovering resource types.
 func GetDiscoveryClient() *discovery.DiscoveryClient {
-	discoveryClient, err := discovery.NewDiscoveryClientForConfig(getKubeConfig())
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(GetKubeConfig())
 	if err != nil {
 		glog.Fatal("Cannot Construct Discovery Client From Config: ", err)
 	}
 	return discoveryClient
 }
 
-func GetKubeClient() *kubernetes.Clientset {
-	kubeClient, err := kubernetes.NewForConfig(getKubeConfig())
-	if err != nil {
-		glog.Fatal("Cannot Construct Kube Client from Config: ", err)
+func GetKubeClient(config *rest.Config) *kubernetes.Clientset {
+	var kubeClient *kubernetes.Clientset
+	var err error
+	if config != nil {
+		kubeClient, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			glog.Fatal("Cannot Construct Kube Client from Config: ", err)
+		}
 	}
+	glog.Error("Cannot Construct Kube Client as input Config is nil")
```

Comment:
This needs to be inside an else or we'll always log it. 
```suggestion
	} else {
	  glog.Error("Cannot Construct Kube Client as input Config is nil")
	}
```

Reply:
Committing because Sherin is unavailable.